analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxTokenListExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Extensions
+{
+    public static class SyntaxTokenListExtensions
+    {
+        public static SyntaxToken? Find(this SyntaxTokenList tokenList, SyntaxKind kind) =>
+            tokenList.IndexOf(kind) is var index && index != -1
```

Comment:
Alternative (I'm not sure whether it is better, so the change is optional):
```suggestion
            tokenList.IndexOf(kind) is >= 0 and var index
```